lib/ruby_lsp/server.rb

```diff
@@ -160,9 +160,10 @@ module RubyLsp
       uri = request.dig(:params, :textDocument, :uri)
 
       Requests::Formatting.new(uri, store.get(uri)).run
-    rescue RuboCop::Runner::InfiniteCorrectionLoop => e
-      show_message(Constant::MessageType::ERROR, "Error from RuboCop: #{e.message}")
-      nil
+    end.on_error do |error|
+      if error.is_a?(RuboCop::Runner::InfiniteCorrectionLoop)
```

Comment:
Should we not handle this error at the `RuboCopRunner` layer?

Reply:
Sorry, that wasn't what I meant to say. I was trying to say that seeing `RuboCop::Runner` here is leaky and might end up leading to bigger failures on projects that don't have RuboCop loaded (our runtime dependency on RuboCop is conditional, remember?).

So, I think the `RuboCopRunner` class should handle this error and maybe raise a `RubyLsp::Requests::Formatting::Error` which can be handled at this layer. We can always get the original RuboCop exception using the `cause`: https://ruby-doc.org/core-3.1.0/Exception.html#cause-method